src/api/c/assign.cpp

```diff
@@ -233,6 +232,11 @@ af_err af_assign_gen(af_array *out,
         af_dtype lhsType= lInfo.getType();
         af_dtype rhsType= rInfo.getType();
 
+        if(rhsDims.ndims() == 0) {
+            dim_t my_dims[] = { 0, 0, 0, 0 };
+            return af_create_handle(out, AF_MAX_DIMS, my_dims, rhsType);
```

Comment:
This is incorrect. We should check if both rhs and index are empty and then return a retained version of `lhs`.